src/models/kv_cache.cpp

```diff
@@ -376,6 +376,33 @@ std::string ComposeKeyValueName(const std::string& template_string, int index) {
   return std::string(key_value_name);
 }
 
+ModelManagedKeyValueCache::ModelManagedKeyValueCache(State& state)
+    : state_{state} {
+  // This fixes with _qa sample where a new instance of ModelManagedKeyValueCache is
+  // created for each new prompt from user. In this case, we need to trigger a KVCache
+  // reset on the session before the first Session::Run.
+  RewindTo(0);
+}
+
+void ModelManagedKeyValueCache::Add() {
+  // NOOP
+}
```

Comment:
nit: unnecessary comment.
```suggestion
void ModelManagedKeyValueCache::Add() {}
```